src/PSR7Client.php

```diff
@@ -53,36 +81,36 @@ public function acceptRequest()
             return null;
         }
 
+        $_SERVER = $this->configureServer($ctx);
+
+        $request = $this->requestFactory->createServerRequest(
+            $ctx['method'],
+            $ctx['uri'],
+            $_SERVER
+        );
+
         parse_str($ctx['rawQuery'], $query);
 
-        $bodyStream = 'php://input';
-        $parsedBody = null;
-        if ($ctx['parsed']) {
-            $parsedBody = json_decode($body, true);
-        } elseif ($body != null) {
-            $bodyStream = new Diactoros\Stream("php://memory", "rwb");
-            $bodyStream->write($body);
+        $request = $request
+            ->withCookieParams($ctx['cookies'])
+            ->withQueryParams($query)
+            ->withUploadedFiles($this->wrapUploads($ctx['uploads']));
+
+        foreach ($ctx['attributes'] as $name => $value) {
+            $request = $request->withAttribute($name, $value);
         }
 
-        $_SERVER = $this->configureServer($ctx);
+        foreach ($ctx['headers'] as $name => $value) {
+            $request = $request->withHeader($name, $value);
+        }
 
-        $request = new Diactoros\ServerRequest(
-            $_SERVER,
-            $this->wrapUploads($ctx['uploads']),
-            $ctx['uri'],
-            $ctx['method'],
-            $bodyStream,
-            $ctx['headers'],
-            $ctx['cookies'],
-            $query,
-            $parsedBody,
-            $ctx['protocol']
-        );
+        if ($ctx['parsed']) {
+            $request = $request->withParsedBody(json_decode($body, true));
+        } else if ($body !== null) {
+            $bodyStream = $this->streamFactory->createStream($body);
```

Comment:
This worries me, you've created stream based on body data and then wrote body into it. Either tests are not complete or json_parser ignore the second segment or something else is wrong.

Reply:
Lol, honest mistake (it mimicked the old code). You are right, this should have broken ~some~ a lot of tests :|